### PR TITLE
[plugin] file management vscode commands

### PR DESCRIPTION
**Pull request by akosyakov**
partially address #4244

You can run vscode-api-tests and observe that tests closing editors after execution.

### CODE REVIEW

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts [hunk 1]
```diff
@@ -54,15 +68,174 @@ export class PluginApiFrontendContribution implements CommandContribution {
...
-        // this command only for compatibility reason
-        commands.registerCommand({ id: 'workbench.action.closeActiveEditor' }, {
-            execute: () => commands.executeCommand('core.close.tab')
-        });
-
         commands.registerCommand(OpenUriCommandHandler.COMMAND_METADATA, {
             execute: (arg: URI) => this.openUriCommandHandler.execute(arg),
             isVisible: () => false
         });
+
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: I thought these commands were going to 
https://github.com/theia-ide/theia/blob/68795ecf308155c85cf6107239201e784976c388/packages/plugin-ext/src/plugin/command-registry.ts#L87

https://github.com/theia-ide/theia/blob/68795ecf308155c85cf6107239201e784976c388/packages/plugin-ext/src/plugin/type-converters.ts#L437

Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: Could you elaborate? I don't know this code. `editor.action.showReferences` is from monaco. Command which I've added they are not contributed by monaco. Also they cannot be implemented in the plugin host process, they need access to the shell and editor manager.

Comment (benoitf) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: https://github.com/theia-ide/theia/pull/4193#issuecomment-459224709

Comment (benoitf) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: it is this comment.

It seems there are different places where there is mapping/aliases between vscode commands and theia commands so might be good to have a single location

Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: I can move it here then: https://github.com/theia-ide/theia/blob/30eaeb02cca616f11d59988ba562eccc3891a6ae/packages/plugin-ext/src/main/browser/command-registry-main.ts#L70

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts [hunk 2]
```diff
@@ -54,15 +68,174 @@ export class PluginApiFrontendContribution implements CommandContribution {
...
+                    });
+                    if (Saveable.is(widget)) {
+                        Saveable.save(widget);
+                    }
+                } else {
+                    this.shell.save();
+                }
+            }
+        });
+        commands.registerCommand({ id: 'workbench.action.files.saveAll', }, {
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: seems to be duplicating 
https://github.com/theia-ide/theia/blob/5cbb4f19d65058f3875fb303ac0ee86a35140491/packages/core/src/browser/common-frontend-contribution.ts#L447-L448

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts [hunk 3]
```diff
@@ -54,15 +68,174 @@ export class PluginApiFrontendContribution implements CommandContribution {
...
+                    });
+                }
+                for (const widget of this.shell.widgets) {
+                    if (NavigatableWidget.is(widget) && widget !== editor) {
+                        widget.close();
+                    }
+                }
+            }
+        });
+        commands.registerCommand({ id: 'workbench.action.closeEditorsInGroup' }, {
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: should these commands around editors be provided by the core instead ? it looks they're not only targeting plug-in system but can be useful for anyone ?

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts (hunk 1 above)
Comment (benoitf) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: I'm fine there or in a command-contribution separate class where we only register these commands like for vscode https://github.com/theia-ide/theia/blob/master/packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts as it's kind easy to know what to find in this class.

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts [hunk 4]
```diff
@@ -33,6 +37,16 @@ export class PluginApiFrontendContribution implements CommandContribution {
...
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    @inject(EditorManager)
+    protected readonly editors: EditorManager;
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: travis-ci doesn't like this consecutive blank line

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts (hunk 3 above)
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: I don't want to mix them:
- `closeEditorsInGroup` would be mean in Theia means different compare to VS Code, in VS Code everything is main area is an editor (webviews, previews and so on), in our case editor is a text editor only
- this commands accept arguments which don't have representation in Theia, like view positions and so on, for now i just did not declare them (so even implementation is not really complete), but it does not mean VS Code extensions cannot pass them

Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: We have close tab command already and you can see that it does completely another thing.

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts (hunk 2 above)
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: For now, yes, i can delegate it. It does not mean that at some point we will need add some custom logic to align with VS Code behaviour.

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts (hunk 1 above)
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: ok plugin-vscode-commands-contribution sounds good as well

File: packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts (hunk 4 above)
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: our linter sometimes is very annoying :)

Comment (benoitf) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: we could add tslint --fix script that fixes automatically tslint errors

Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: let's discuss next week on dev meeting? i guess it should be git hook or something like that

Comment (benoitf) [reply packages/plugin-ext/src/main/browser/plugin-frontend-contribution.ts]: you can discuss it but I'm on PTO next week and at RHT summit the week after so I may not be able to attend the dev meeting.

**Comment by akosyakov** [comment]
@benoitf i've moved commands to vscode extension and included launch config for plugin api tests that we don't need to copy it each time.

**Pull request merged**